File: backend/messager/src/domain/username.ts
export default class Username {
	constructor(readonly value: string) { 
		if (!this.isValidLength(value)) throw new Error('Username must contain at least 4 characters')
	}

	private isValidLength(username: string) {
		if (username.length < 4) return false
		return true
	}
}

File: backend/auth/auth/domain/entities/username.cs
public class Username
{
    public string value {get;}
    public Username(string value){
        if(!this.isValidLength(value)) throw new ArgumentException("Username must contain at least 4 characters");
        this.value = value;
    }

    private bool isValidLength(string value) {
        if (value.Length < 4) return false;
        return true;
    }
}